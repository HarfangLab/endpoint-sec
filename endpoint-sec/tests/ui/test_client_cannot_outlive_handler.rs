use endpoint_sec::Client;

fn main() {
    let mut client = {
        let v = vec![1, 2, 3];
        let handler = |_: &mut Client<'_>, _| {
            dbg!(&v);
        };
        let client = Client::new(handler).unwrap();

        std::mem::drop(handler);
        client
    };
    client.unsubscribe_all();
}
