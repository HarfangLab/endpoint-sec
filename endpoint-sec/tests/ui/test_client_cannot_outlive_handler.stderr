error[E0597]: `v` does not live long enough
  --> tests/ui/test_client_cannot_outlive_handler.rs:7:19
   |
4  |     let mut client = {
   |         ---------- borrow later stored here
5  |         let v = vec![1, 2, 3];
6  |         let handler = |_: &mut Client<'_>, _| {
   |                       ----------------------- value captured here
7  |             dbg!(&v);
   |                   ^ borrowed value does not live long enough
...
13 |     };
   |     - `v` dropped here while still borrowed
